src/operator/c_lapack_api.cc

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "c_lapack_api.h"
+
+#if (MSHADOW_USE_MKL && MXNET_USE_LAPACK)
```

Comment:
Why do we need the left hand side of this &&, MSHADOW_USE_MKL if we're testing against only the right hand side in the next line?  For what cases would line 22 alone be true and line 23 + line 22 be false?

Reply:
So it's the same logic in the header in line 153:  https://github.com/apache/incubator-mxnet/pull/13535/files#diff-3600054dbf1ab219ca274dc6f21869b5R153

Reply:
I don't see that it's the same, maybe I'm missing something.  Here's what doesn't make sense to me:

```cpp
#if (MSHADOW_USE_MKL && MXNET_USE_LAPACK)
#elif MXNET_USE_LAPACK
#else 
do_something...
```
Looking at this with a simple truth table:

Assume MXNET_USE_LAPACK is true, Assume MSHADOW_USE_MKL true -> Do nothing
Assume MXNET_USE_LAPACK is true, Assume MSHADOW_USE_MKL false -> Do nothing
Assume MXNET_USE_LAPACK is false, Assume MSHADOW_USE_MKL true -> Do something
Assume MXNET_USE_LAPACK is false, Assume MSHADOW_USE_MKL false -> Do something

So no matter what we set MSHADOW_USE_MKL to it won't affect the control flow right? (or am I missing something really basic which is totally possible).

Reply:
My point is that it's matching the define flow from the header file. But since it's causing confusion and I think your analysis is right, I can change it to read
```
 #if MXNET_USE_LAPACK
  ....
 #endif 

```
From your commend I understand that this would be your preference, am I correct?

Reply:
Right, I just think it better describes what's going on. Thanks Pedro.